CRM/EventsExtras/Hook/BuildForm.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use CRM_EventsExtras_ExtensionUtil as E;
+use CRM_EventsExtras_SettingsManager as SettingsManager;
+
+/**
+ * Abstract class for BuildForm Hook
+ */
+abstract class CRM_EventsExtras_Hook_BuildForm {
```

Comment:
I think a better name will be `CRM_EventsExtras_Hook_BuildForm_BaseEvent`  or something similar
Since it is a class that other Event build form Hook classes extend from.